fly/src/sorts.rs

```diff
@@ -281,21 +167,52 @@ pub fn term_has_all_sort_annotations(term: &Term) -> bool {
             binders
                 .iter()
                 .all(|binder| binder.sort != Sort::Id("".to_owned()))
-                && term_has_all_sort_annotations(body)
+                && has_all_sort_annotations_term(body)
         }
     }
 }
 
-// can either hold a function sort or the index of a sort variable
-// all sorts must be known by the time `check` returns
+/// Represents the sort of a name, which can either name a relation or an individual.
+/// Relations (including functions) always take at least one argument.
+/// Individuals can additionally have "unknown" sort, which will be determined during sort inference.
 #[derive(Clone, Debug)]
-enum NamedSort {
-    Known(Vec<Sort>, Sort),
-    Unknown(SortVar),
+enum RelationOrIndividual {
```

Comment:
We talked about this in person, but just for the record, we decided to keep this change because the sort checker reports and error if you use an App with zero arguments.